lib/common/imageTool.py

```diff
@@ -124,15 +129,132 @@ def convert_to_dct(self, image_fp):
+        print result_list
+        return result_list
+
+    def template_match(self, base_img_fp, template_fp):
+        img = cv2.imread(base_img_fp)
+        img_gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        template = cv2.imread(template_fp, 0)
+        w, h = template.shape[::-1]
+
+        #Choose SQDIFF_NORMED method to perform template matching
+        methods = 'cv2.TM_SQDIFF_NORMED'
+        method_eval = eval(methods)
+
+        # Apply template Matching
+        res = cv2.matchTemplate(img_gray, template, method_eval)
+        min_val, max_val, min_loc, max_loc = cv2.minMaxLoc(res)
+
+        #For artifactual image output
+        '''
+        str_image_fp = $TBD
+        #Draw a rectangle on the target object
+        top_left = min_loc
+        bottom_right = (top_left[0] + w, top_left[1] + h)
+        cv2.imwrite(str_image_fp, cv2.rectangle(img, top_left, bottom_right, (0, 0, 255), 2))
+
+        #cv2.imshow('original', img)
+        #cv2.waitKey(0)
+        '''
+        return min_val
+
+    def crop_image(self, input_sample_fp, output_sample_fp, coord=[]):
+        img = cv2.imread(input_sample_fp)
+        img_gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        w, h = img_gray.shape[::-1]
+        if not coord or len(coord) != 2 or \
+                        type(coord[0]) is not tuple or len(coord[0]) !=2 or \
+                        type(coord[1]) is not tuple or len(coord[1]) !=2 :
+            coord = [(0,0), (w,h)]
+            print "WARNING: Incorrect coordinates, using fully image crop"
+        else:
+            if coord[0][0] < 0:
+                coord[0] = (0, coord[0][1])
+                print "WARNING: Incorrect coordinates, set origin x coordinate to 0"
+            if coord[0][0] > w:
+                coord[0] = (w, coord[0][1])
+                print "WARNING: Incorrect coordinates, set origin x coordinate to width"
+            if coord[1][0] < 0:
+                coord[1] = (0, coord[1][1])
+                print "WARNING: Incorrect coordinates, set target x coordinate to 0"
+            if coord[1][0] > w:
+                coord[1] = (w, coord[1][1])
+                print "WARNING: Incorrect coordinates, set target x coordinate to width"
+            if coord[0][1] < 0:
+                coord[0] = (coord[0][0], 0)
+                print "WARNING: Incorrect coordinates, set origin y coordinate to 0"
+            if coord[0][1] > h:
+                coord[0] = (coord[0][0], h)
+                print "WARNING: Incorrect coordinates, set origin y coordinate to height"
+            if coord[1][1] < 0:
+                coord[1] = (coord[1][0], 0)
+                print "WARNING: Incorrect coordinates, set target y coordinate to 0"
+            if coord[1][1] > h:
+                coord[1] = (coord[1][0], h)
+                print "WARNING: Incorrect coordinates, set target y coordinate to height"
+        print "Crop image range" + str(coord)
```

Comment:
you can simplify the login, for example:

for i in range(2):
    for j in range(2):
        new_val = coord[i][j]
        print coord[i][j]
        if coord[i][j] < 0:
            new_val = 0
        elif j == 0 and coord[i][j] > w:
            new_val = w
        elif j == 1 and coord[i][j] > h:
            new_val = h
        print new_val
        if new_val != coord[i][j]:
            new_xy = [coord[i][0], coord[i][1]]
            list_index = int(j == 0)
            new_xy[list_index] = coord[i][list_index]
            new_xy[j] = new_val
            coord[i] = tuple(new_xy)
            print "WARNING: Incorrect coordinates, set %s %s coordinate to %s" % (["origin", "target"][i],str(unichr(120+j)), str(new_val))